### PR TITLE
updateStrategy and Pod scheduling changes

### PR DESCRIPTION
**Pull request by ambiknai**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] I have made sure existing UTs are not impacted
- [ ] I have executed all necessary linter and made sure internal pipeline/travis is not broken

### CODE REVIEW

**Comment by ambiknai** [comment]
Test Results
```
ginkgo -v -focus="s3fs" -debug  ./tests/e2e 
  I0710 04:48:29.930886 503469 test_context.go:561] The --provider flag is not set. Continuing as if --provider=skeleton had been used.
Running Suite: Object Storage CSI Driver End-to-End Tests - /root/workspace/src/github.com/IBM/ibm-object-csi-driver/tests/e2e
==============================================================================================================================
Random Seed: 1720612102

Will run 1 of 2 specs
S
------------------------------
s3fs with s3fs SC, Create a volume - Attach to a Pod  - Read/Write
/root/workspace/src/github.com/IBM/ibm-object-csi-driver/tests/e2e/s3fs_test.go:137
  STEP: Creating a kubernetes client @ 07/10/24 04:48:29.931
  I0710 04:48:29.931779 503469 util.go:506] >>> kubeConfig: /root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml
  STEP: Building a namespace api object, basename obj-e2e-s3fs @ 07/10/24 04:48:29.933
  STEP: Waiting for a default service account to be provisioned in namespace @ 07/10/24 04:48:30.902
  STEP: Waiting for kube-root-ca.crt to be provisioned in namespace @ 07/10/24 04:48:31.209
  STEP: creating Secret @ 07/10/24 04:48:31.681
  I0710 04:48:31.681467 503469 basespecs.go:116] creating Secret "obj-e2e-s3fs-2953" under "obj-e2e-s3fs-2953"
  STEP: setting up POD @ 07/10/24 04:48:31.842
  STEP: setting up the PVC for POD @ 07/10/24 04:48:31.842
  STEP: setting up the PVC and PV @ 07/10/24 04:48:31.842
  STEP: creating a PVC @ 07/10/24 04:48:31.842
  STEP: waiting for PVC to be in phase "Bound" @ 07/10/24 04:48:32.176
  I0710 04:48:32.176858 503469 pv.go:780] Waiting up to timeout=5m0s for PersistentVolumeClaims [obj-e2e-s3fs-2953] to have phase Bound
  I0710 04:48:32.335327 503469 pv.go:796] PersistentVolumeClaim obj-e2e-s3fs-2953 found but phase is Pending instead of Bound.
  I0710 04:48:34.497550 503469 pv.go:791] PersistentVolumeClaim obj-e2e-s3fs-2953 found and phase=Bound (2.320603842s)
  STEP: checking the PVC @ 07/10/24 04:48:34.497
  STEP: validating provisioned PV @ 07/10/24 04:48:34.657
  I0710 04:48:34.818102 503469 basespecs.go:295] validating provisioned PV [pvc-6666045c-4b97-4328-8157-ef97e961fc01] for PVC [obj-e2e-s3fs-2953]
  STEP: checking PV @ 07/10/24 04:48:34.818
  I0710 04:48:34.818257 503469 basespecs.go:307] checking PV [pvc-6666045c-4b97-4328-8157-ef97e961fc01]
  STEP: deploying the pod "0" @ 07/10/24 04:48:34.818
  W0710 04:48:34.984013  503469 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "obj-e2e-tester" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "obj-e2e-tester" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "obj-e2e-tester" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "obj-e2e-tester" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  STEP: checking that the pods status is running @ 07/10/24 04:48:34.984
  STEP: checking that the pods status is running [obj-e2e-s3fs-2953/obj-e2e-tester-kpt6t] @ 07/10/24 04:48:34.984
  STEP: Saw pod running @ 07/10/24 04:48:43.802
  STEP: checking pod exec after pod recreate @ 07/10/24 04:48:43.802
  STEP: POD Exec: executing cmd in POD @ 07/10/24 04:48:43.802
  I0710 04:48:43.803000 503469 basespecs.go:162] executing cmd in POD [obj-e2e-s3fs-2953/obj-e2e-tester-kpt6t]
  I0710 04:48:43.803101 503469 builder.go:121] Running '/usr/local/bin/kubectl --kubeconfig=/root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml --namespace=obj-e2e-s3fs-2953 exec obj-e2e-tester-kpt6t --namespace=obj-e2e-s3fs-2953 -- cat /mnt/test-1/data'
  I0710 04:48:46.330763 503469 builder.go:146] stderr: ""
  I0710 04:48:46.330874 503469 builder.go:147] stdout: "hello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\nhello world\n"
  STEP: POD Cleanup: deleting POD @ 07/10/24 04:48:46.33
  I0710 04:48:46.330982 503469 basespecs.go:350] deleting POD [obj-e2e-s3fs-2953/obj-e2e-tester-kpt6t]
  I0710 04:48:46.493849 503469 builder.go:121] Running '/usr/local/bin/kubectl --kubeconfig=/root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml --namespace=obj-e2e-s3fs-2953 describe po obj-e2e-tester-kpt6t'
  I0710 04:48:47.386323 503469 builder.go:146] stderr: ""
  I0710 04:48:47.386451 503469 builder.go:147] stdout: "Name:             obj-e2e-tester-kpt6t\nNamespace:        obj-e2e-s3fs-2953\nPriority:         0\nService Account:  default\nNode:             10.245.0.15/10.245.0.15\nStart Time:       Wed, 10 Jul 2024 04:48:34 -0700\nLabels:           app=obj-vol-e2e\nAnnotations:      cni.projectcalico.org/containerID: 57aa39bf08459fb2f5619a694cfa8558a3fab111e79875f4bbe8c1f67e95d2b1\n                  cni.projectcalico.org/podIP: 172.17.113.7/32\n                  cni.projectcalico.org/podIPs: 172.17.113.7/32\nStatus:           Running\nIP:               172.17.113.7\nIPs:\n  IP:  172.17.113.7\nContainers:\n  obj-e2e-tester:\n    Container ID:  containerd://8f98ddc90c89b8ff97151981acc9983d4976014749cfab361a35f44232db9003\n    Image:         registry.k8s.io/e2e-test-images/busybox:1.36.1-1\n    Image ID:      registry.k8s.io/e2e-test-images/busybox@sha256:a9155b13325b2abef48e71de77bb8ac015412a566829f621d06bfae5c699b1b9\n    Port:          <none>\n    Host Port:     <none>\n    Command:\n      /bin/sh\n    Args:\n      -c\n      echo 'hello world' >> /mnt/test-1/data && while true; do sleep 2; done\n    State:          Running\n      Started:      Wed, 10 Jul 2024 04:48:42 -0700\n    Ready:          True\n    Restart Count:  0\n    Environment:    <none>\n    Mounts:\n      /mnt/test-1 from test-volume-1 (rw)\n      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-zwsvh (ro)\nConditions:\n  Type                        Status\n  PodReadyToStartContainers   True \n  Initialized                 True \n  Ready                       True \n  ContainersReady             True \n  PodScheduled                True \nVolumes:\n  test-volume-1:\n    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)\n    ClaimName:  obj-e2e-s3fs-2953\n    ReadOnly:   false\n  kube-api-access-zwsvh:\n    Type:                    Projected (a volume that contains injected data from multiple sources)\n    TokenExpirationSeconds:  3607\n    ConfigMapName:           kube-root-ca.crt\n    ConfigMapOptional:       <nil>\n    DownwardAPI:             true\nQoS Class:                   BestEffort\nNode-Selectors:              <none>\nTolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 600s\n                             node.kubernetes.io/unreachable:NoExecute op=Exists for 600s\nEvents:\n  Type    Reason     Age   From               Message\n  ----    ------     ----  ----               -------\n  Normal  Scheduled  13s   default-scheduler  Successfully assigned obj-e2e-s3fs-2953/obj-e2e-tester-kpt6t to 10.245.0.15\n  Normal  Pulling    10s   kubelet            Pulling image \"registry.k8s.io/e2e-test-images/busybox:1.36.1-1\"\n  Normal  Pulled     5s    kubelet            Successfully pulled image \"registry.k8s.io/e2e-test-images/busybox:1.36.1-1\" in 4.543s (4.543s including waiting). Image size: 2223659 bytes.\n  Normal  Created    5s    kubelet            Created container obj-e2e-tester\n  Normal  Started    5s    kubelet            Started container obj-e2e-tester\n"
  I0710 04:48:47.386514 503469 output.go:144] 
  Output of kubectl describe obj-e2e-tester-kpt6t:
  Name:             obj-e2e-tester-kpt6t
  Namespace:        obj-e2e-s3fs-2953
  Priority:         0
  Service Account:  default
  Node:             10.245.0.15/10.245.0.15
  Start Time:       Wed, 10 Jul 2024 04:48:34 -0700
  Labels:           app=obj-vol-e2e
  Annotations:      cni.projectcalico.org/containerID: 57aa39bf08459fb2f5619a694cfa8558a3fab111e79875f4bbe8c1f67e95d2b1
                    cni.projectcalico.org/podIP: 172.17.113.7/32
                    cni.projectcalico.org/podIPs: 172.17.113.7/32
  Status:           Running
  IP:               172.17.113.7
  IPs:
    IP:  172.17.113.7
  Containers:
    obj-e2e-tester:
      Container ID:  containerd://8f98ddc90c89b8ff97151981acc9983d4976014749cfab361a35f44232db9003
      Image:         registry.k8s.io/e2e-test-images/busybox:1.36.1-1
      Image ID:      registry.k8s.io/e2e-test-images/busybox@sha256:a9155b13325b2abef48e71de77bb8ac015412a566829f621d06bfae5c699b1b9
      Port:          <none>
      Host Port:     <none>
      Command:
        /bin/sh
      Args:
        -c
        echo 'hello world' >> /mnt/test-1/data && while true; do sleep 2; done
      State:          Running
        Started:      Wed, 10 Jul 2024 04:48:42 -0700
      Ready:          True
      Restart Count:  0
      Environment:    <none>
      Mounts:
        /mnt/test-1 from test-volume-1 (rw)
        /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-zwsvh (ro)
  Conditions:
    Type                        Status
    PodReadyToStartContainers   True 
    Initialized                 True 
    Ready                       True 
    ContainersReady             True 
    PodScheduled                True 
  Volumes:
    test-volume-1:
      Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
      ClaimName:  obj-e2e-s3fs-2953
      ReadOnly:   false
    kube-api-access-zwsvh:
      Type:                    Projected (a volume that contains injected data from multiple sources)
      TokenExpirationSeconds:  3607
      ConfigMapName:           kube-root-ca.crt
      ConfigMapOptional:       <nil>
      DownwardAPI:             true
  QoS Class:                   BestEffort
  Node-Selectors:              <none>
  Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 600s
                               node.kubernetes.io/unreachable:NoExecute op=Exists for 600s
  Events:
    Type    Reason     Age   From               Message
    ----    ------     ----  ----               -------
    Normal  Scheduled  13s   default-scheduler  Successfully assigned obj-e2e-s3fs-2953/obj-e2e-tester-kpt6t to 10.245.0.15
    Normal  Pulling    10s   kubelet            Pulling image "registry.k8s.io/e2e-test-images/busybox:1.36.1-1"
    Normal  Pulled     5s    kubelet            Successfully pulled image "registry.k8s.io/e2e-test-images/busybox:1.36.1-1" in 4.543s (4.543s including waiting). Image size: 2223659 bytes.
    Normal  Created    5s    kubelet            Created container obj-e2e-tester
    Normal  Started    5s    kubelet            Started container obj-e2e-tester

  I0710 04:48:47.387495 503469 builder.go:121] Running '/usr/local/bin/kubectl --kubeconfig=/root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml --namespace=obj-e2e-s3fs-2953 logs obj-e2e-tester-kpt6t --tail=100'
  I0710 04:48:48.151891 503469 builder.go:146] stderr: ""
  I0710 04:48:48.151995 503469 builder.go:147] stdout: ""
  I0710 04:48:48.152022 503469 output.go:147] 
  Last 100 log lines of obj-e2e-tester-kpt6t:

  I0710 04:48:48.374264 503469 basespecs.go:363] Pod obj-e2e-tester-kpt6t has the following logs: 
  I0710 04:48:48.374355 503469 basespecs.go:366] deleting POD [obj-e2e-s3fs-2953/obj-e2e-tester-kpt6t]
  STEP: Deleting pod obj-e2e-tester-kpt6t in namespace obj-e2e-s3fs-2953 @ 07/10/24 04:48:48.374
  STEP: deleting PVC [obj-e2e-s3fs-2953] @ 07/10/24 04:48:48.549
  I0710 04:48:48.549389 503469 pv.go:201] Deleting PersistentVolumeClaim "obj-e2e-s3fs-2953"
  STEP: Triggered DeletePersistentVolumeClaim call @ 07/10/24 04:48:48.717
  I0710 04:48:48.717343 503469 basespecs.go:263] deleting PVC [obj-e2e-s3fs-2953/obj-e2e-s3fs-2953] using PV [pvc-6666045c-4b97-4328-8157-ef97e961fc01]
  STEP: waiting for claim's PV [pvc-6666045c-4b97-4328-8157-ef97e961fc01] to be deleted @ 07/10/24 04:48:48.717
  I0710 04:48:48.717406 503469 pv.go:853] Waiting up to 10m0s for PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 to get deleted
  I0710 04:48:48.880348 503469 pv.go:857] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 found and phase=Bound (162.875318ms)
  I0710 04:48:54.042019 503469 pv.go:857] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 found and phase=Bound (5.324548486s)
  I0710 04:48:59.204806 503469 pv.go:857] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 found and phase=Bound (10.48733258s)
  I0710 04:49:04.367350 503469 pv.go:857] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 found and phase=Bound (15.649880496s)
  I0710 04:49:09.532960 503469 pv.go:857] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 found and phase=Bound (20.815491147s)
  I0710 04:49:14.695893 503469 pv.go:857] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 found and phase=Bound (25.978420655s)
  I0710 04:49:19.858309 503469 pv.go:861] PersistentVolume pvc-6666045c-4b97-4328-8157-ef97e961fc01 was removed
Test passed
  STEP: deleting Secret @ 07/10/24 04:49:19.858
  I0710 04:49:19.858578 503469 basespecs.go:140] deleting Secret [obj-e2e-s3fs-2953]
  STEP: Destroying namespace "obj-e2e-s3fs-2953" for this suite. @ 07/10/24 04:49:20.02
• [50.259 seconds]
------------------------------

Ran 1 of 2 Specs in 50.260 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 1 Skipped
You're using deprecated Ginkgo functionality:
=============================================
  --ginkgo.slow-spec-threshold is deprecated --slow-spec-threshold has been deprecated and will be removed in a future version of Ginkgo.  This feature has proved to be more noisy than useful.  You can use --poll-progress-after, instead, to get more actionable feedback about potentially slow specs and understand where they might be getting stuck.

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=2.19.0

PASS

Ginkgo ran 1 suite in 57.371922056s
Test Suite Passed
You're using deprecated Ginkgo functionality:
=============================================
  --debug is deprecated
  Learn more at: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#removed--debug

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=2.1.6

root@devVm1:~/workspace/src/github.com/IBM/ibm-object-csi-driver# export rcloneBucketName=testbuckete3temp
root@devVm1:~/workspace/src/github.com/IBM/ibm-object-csi-driver# 
root@devVm1:~/workspace/src/github.com/IBM/ibm-object-csi-driver# 
root@devVm1:~/workspace/src/github.com/IBM/ibm-object-csi-driver# 
root@devVm1:~/workspace/src/github.com/IBM/ibm-object-csi-driver# ginkgo -v -focus="rclone" -debug  ./tests/e2e 
  I0710 04:49:42.221420 503682 test_context.go:561] The --provider flag is not set. Continuing as if --provider=skeleton had been used.
Running Suite: Object Storage CSI Driver End-to-End Tests - /root/workspace/src/github.com/IBM/ibm-object-csi-driver/tests/e2e
==============================================================================================================================
Random Seed: 1720612176

Will run 1 of 2 specs
S
------------------------------
rclone with rclone SC, Create a volume - Attach to a Pod  - Read/Write
/root/workspace/src/github.com/IBM/ibm-object-csi-driver/tests/e2e/s3fs_test.go:62
  STEP: Creating a kubernetes client @ 07/10/24 04:49:42.222
  I0710 04:49:42.222322 503682 util.go:506] >>> kubeConfig: /root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml
  STEP: Building a namespace api object, basename obj-e2e-rclone @ 07/10/24 04:49:42.224
  STEP: Waiting for a default service account to be provisioned in namespace @ 07/10/24 04:49:44.127
  STEP: Waiting for kube-root-ca.crt to be provisioned in namespace @ 07/10/24 04:49:44.505
  STEP: creating Secret @ 07/10/24 04:49:45.081
  I0710 04:49:45.081512 503682 basespecs.go:116] creating Secret "obj-e2e-rclone-6849" under "obj-e2e-rclone-6849"
  STEP: setting up POD @ 07/10/24 04:49:45.275
  STEP: setting up the PVC for POD @ 07/10/24 04:49:45.276
  STEP: setting up the PVC and PV @ 07/10/24 04:49:45.276
  STEP: creating a PVC @ 07/10/24 04:49:45.276
  STEP: waiting for PVC to be in phase "Bound" @ 07/10/24 04:49:45.674
  I0710 04:49:45.674759 503682 pv.go:780] Waiting up to timeout=5m0s for PersistentVolumeClaims [obj-e2e-rclone-6849] to have phase Bound
  I0710 04:49:45.870741 503682 pv.go:796] PersistentVolumeClaim obj-e2e-rclone-6849 found but phase is Pending instead of Bound.
  I0710 04:49:48.067658 503682 pv.go:791] PersistentVolumeClaim obj-e2e-rclone-6849 found and phase=Bound (2.392835438s)
  STEP: checking the PVC @ 07/10/24 04:49:48.067
  STEP: validating provisioned PV @ 07/10/24 04:49:48.263
  I0710 04:49:48.460454 503682 basespecs.go:295] validating provisioned PV [pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8] for PVC [obj-e2e-rclone-6849]
  STEP: checking PV @ 07/10/24 04:49:48.46
  I0710 04:49:48.460594 503682 basespecs.go:307] checking PV [pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8]
  STEP: deploying the pod "0" @ 07/10/24 04:49:48.46
  W0710 04:49:48.664737  503682 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "obj-e2e-tester" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "obj-e2e-tester" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "obj-e2e-tester" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "obj-e2e-tester" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  STEP: checking that the pods status is running @ 07/10/24 04:49:48.664
  STEP: checking that the pods status is running [obj-e2e-rclone-6849/obj-e2e-tester-m4tkh] @ 07/10/24 04:49:48.665
  STEP: Saw pod running @ 07/10/24 04:49:51.059
  STEP: checking pod exec after pod recreate @ 07/10/24 04:49:51.059
  STEP: POD Exec: executing cmd in POD @ 07/10/24 04:49:51.059
  I0710 04:49:51.059565 503682 basespecs.go:162] executing cmd in POD [obj-e2e-rclone-6849/obj-e2e-tester-m4tkh]
  I0710 04:49:51.059662 503682 builder.go:121] Running '/usr/local/bin/kubectl --kubeconfig=/root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml --namespace=obj-e2e-rclone-6849 exec obj-e2e-tester-m4tkh --namespace=obj-e2e-rclone-6849 -- cat /mnt/test-1/data'
  I0710 04:49:53.245604 503682 builder.go:146] stderr: ""
  I0710 04:49:53.245696 503682 builder.go:147] stdout: "hello world\n"
  STEP: POD Cleanup: deleting POD @ 07/10/24 04:49:53.245
  I0710 04:49:53.245794 503682 basespecs.go:350] deleting POD [obj-e2e-rclone-6849/obj-e2e-tester-m4tkh]
  I0710 04:49:53.443792 503682 builder.go:121] Running '/usr/local/bin/kubectl --kubeconfig=/root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml --namespace=obj-e2e-rclone-6849 describe po obj-e2e-tester-m4tkh'
  I0710 04:49:54.543195 503682 builder.go:146] stderr: ""
  I0710 04:49:54.543326 503682 builder.go:147] stdout: "Name:             obj-e2e-tester-m4tkh\nNamespace:        obj-e2e-rclone-6849\nPriority:         0\nService Account:  default\nNode:             10.245.128.13/10.245.128.13\nStart Time:       Wed, 10 Jul 2024 04:49:48 -0700\nLabels:           app=obj-vol-e2e\nAnnotations:      cni.projectcalico.org/containerID: 788d4795773a39d3d87d3405d9fe273ba75069aedd49b7a930632b5da8e99704\n                  cni.projectcalico.org/podIP: 172.17.114.11/32\n                  cni.projectcalico.org/podIPs: 172.17.114.11/32\nStatus:           Running\nIP:               172.17.114.11\nIPs:\n  IP:  172.17.114.11\nContainers:\n  obj-e2e-tester:\n    Container ID:  containerd://fb54478051988b399624cad83ce06104c9e873c75fc0572756d52a6e0d2880dd\n    Image:         registry.k8s.io/e2e-test-images/busybox:1.36.1-1\n    Image ID:      registry.k8s.io/e2e-test-images/busybox@sha256:a9155b13325b2abef48e71de77bb8ac015412a566829f621d06bfae5c699b1b9\n    Port:          <none>\n    Host Port:     <none>\n    Command:\n      /bin/sh\n    Args:\n      -c\n      echo 'hello world' >> /mnt/test-1/data && while true; do sleep 2; done\n    State:          Running\n      Started:      Wed, 10 Jul 2024 04:49:49 -0700\n    Ready:          True\n    Restart Count:  0\n    Environment:    <none>\n    Mounts:\n      /mnt/test-1 from test-volume-1 (rw)\n      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8kh8l (ro)\nConditions:\n  Type                        Status\n  PodReadyToStartContainers   True \n  Initialized                 True \n  Ready                       True \n  ContainersReady             True \n  PodScheduled                True \nVolumes:\n  test-volume-1:\n    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)\n    ClaimName:  obj-e2e-rclone-6849\n    ReadOnly:   false\n  kube-api-access-8kh8l:\n    Type:                    Projected (a volume that contains injected data from multiple sources)\n    TokenExpirationSeconds:  3607\n    ConfigMapName:           kube-root-ca.crt\n    ConfigMapOptional:       <nil>\n    DownwardAPI:             true\nQoS Class:                   BestEffort\nNode-Selectors:              <none>\nTolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 600s\n                             node.kubernetes.io/unreachable:NoExecute op=Exists for 600s\nEvents:\n  Type    Reason     Age   From               Message\n  ----    ------     ----  ----               -------\n  Normal  Scheduled  6s    default-scheduler  Successfully assigned obj-e2e-rclone-6849/obj-e2e-tester-m4tkh to 10.245.128.13\n  Normal  Pulled     5s    kubelet            Container image \"registry.k8s.io/e2e-test-images/busybox:1.36.1-1\" already present on machine\n  Normal  Created    5s    kubelet            Created container obj-e2e-tester\n  Normal  Started    5s    kubelet            Started container obj-e2e-tester\n"
  I0710 04:49:54.543378 503682 output.go:144] 
  Output of kubectl describe obj-e2e-tester-m4tkh:
  Name:             obj-e2e-tester-m4tkh
  Namespace:        obj-e2e-rclone-6849
  Priority:         0
  Service Account:  default
  Node:             10.245.128.13/10.245.128.13
  Start Time:       Wed, 10 Jul 2024 04:49:48 -0700
  Labels:           app=obj-vol-e2e
  Annotations:      cni.projectcalico.org/containerID: 788d4795773a39d3d87d3405d9fe273ba75069aedd49b7a930632b5da8e99704
                    cni.projectcalico.org/podIP: 172.17.114.11/32
                    cni.projectcalico.org/podIPs: 172.17.114.11/32
  Status:           Running
  IP:               172.17.114.11
  IPs:
    IP:  172.17.114.11
  Containers:
    obj-e2e-tester:
      Container ID:  containerd://fb54478051988b399624cad83ce06104c9e873c75fc0572756d52a6e0d2880dd
      Image:         registry.k8s.io/e2e-test-images/busybox:1.36.1-1
      Image ID:      registry.k8s.io/e2e-test-images/busybox@sha256:a9155b13325b2abef48e71de77bb8ac015412a566829f621d06bfae5c699b1b9
      Port:          <none>
      Host Port:     <none>
      Command:
        /bin/sh
      Args:
        -c
        echo 'hello world' >> /mnt/test-1/data && while true; do sleep 2; done
      State:          Running
        Started:      Wed, 10 Jul 2024 04:49:49 -0700
      Ready:          True
      Restart Count:  0
      Environment:    <none>
      Mounts:
        /mnt/test-1 from test-volume-1 (rw)
        /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8kh8l (ro)
  Conditions:
    Type                        Status
    PodReadyToStartContainers   True 
    Initialized                 True 
    Ready                       True 
    ContainersReady             True 
    PodScheduled                True 
  Volumes:
    test-volume-1:
      Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
      ClaimName:  obj-e2e-rclone-6849
      ReadOnly:   false
    kube-api-access-8kh8l:
      Type:                    Projected (a volume that contains injected data from multiple sources)
      TokenExpirationSeconds:  3607
      ConfigMapName:           kube-root-ca.crt
      ConfigMapOptional:       <nil>
      DownwardAPI:             true
  QoS Class:                   BestEffort
  Node-Selectors:              <none>
  Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 600s
                               node.kubernetes.io/unreachable:NoExecute op=Exists for 600s
  Events:
    Type    Reason     Age   From               Message
    ----    ------     ----  ----               -------
    Normal  Scheduled  6s    default-scheduler  Successfully assigned obj-e2e-rclone-6849/obj-e2e-tester-m4tkh to 10.245.128.13
    Normal  Pulled     5s    kubelet            Container image "registry.k8s.io/e2e-test-images/busybox:1.36.1-1" already present on machine
    Normal  Created    5s    kubelet            Created container obj-e2e-tester
    Normal  Started    5s    kubelet            Started container obj-e2e-tester

  I0710 04:49:54.544382 503682 builder.go:121] Running '/usr/local/bin/kubectl --kubeconfig=/root/.bluemix/plugins/container-service/clusters/amb-cos-test-cq76dghs0vk1qkuubd3g-admin/kube-config.yaml --namespace=obj-e2e-rclone-6849 logs obj-e2e-tester-m4tkh --tail=100'
  I0710 04:49:55.470406 503682 builder.go:146] stderr: ""
  I0710 04:49:55.470484 503682 builder.go:147] stdout: ""
  I0710 04:49:55.470505 503682 output.go:147] 
  Last 100 log lines of obj-e2e-tester-m4tkh:

  I0710 04:49:55.708789 503682 basespecs.go:363] Pod obj-e2e-tester-m4tkh has the following logs: 
  I0710 04:49:55.708874 503682 basespecs.go:366] deleting POD [obj-e2e-rclone-6849/obj-e2e-tester-m4tkh]
  STEP: Deleting pod obj-e2e-tester-m4tkh in namespace obj-e2e-rclone-6849 @ 07/10/24 04:49:55.708
  STEP: deleting PVC [obj-e2e-rclone-6849] @ 07/10/24 04:49:55.92
  I0710 04:49:55.920501 503682 pv.go:201] Deleting PersistentVolumeClaim "obj-e2e-rclone-6849"
  STEP: Triggered DeletePersistentVolumeClaim call @ 07/10/24 04:49:56.125
  I0710 04:49:56.125189 503682 basespecs.go:263] deleting PVC [obj-e2e-rclone-6849/obj-e2e-rclone-6849] using PV [pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8]
  STEP: waiting for claim's PV [pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8] to be deleted @ 07/10/24 04:49:56.125
  I0710 04:49:56.125249 503682 pv.go:853] Waiting up to 10m0s for PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 to get deleted
  I0710 04:49:56.323105 503682 pv.go:857] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 found and phase=Bound (197.816611ms)
  I0710 04:50:01.520130 503682 pv.go:857] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 found and phase=Bound (5.394845516s)
  I0710 04:50:06.715980 503682 pv.go:857] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 found and phase=Bound (10.590695016s)
  I0710 04:50:12.495542 503682 pv.go:857] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 found and phase=Bound (16.370258304s)
  I0710 04:50:18.279813 503682 pv.go:857] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 found and phase=Bound (22.154531197s)
  I0710 04:50:23.478352 503682 pv.go:857] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 found and phase=Bound (27.353069378s)
  I0710 04:50:28.675142 503682 pv.go:861] PersistentVolume pvc-caf5d45f-abfa-4cc7-b46f-e2c0a9eb7eb8 was removed
Test passed
  STEP: deleting Secret @ 07/10/24 04:50:28.675
  I0710 04:50:28.675358 503682 basespecs.go:140] deleting Secret [obj-e2e-rclone-6849]
  STEP: Destroying namespace "obj-e2e-rclone-6849" for this suite. @ 07/10/24 04:50:28.871
• [46.847 seconds]
------------------------------

Ran 1 of 2 Specs in 46.847 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 1 Skipped
PASS
You're using deprecated Ginkgo functionality:
=============================================
  --ginkgo.slow-spec-threshold is deprecated --slow-spec-threshold has been deprecated and will be removed in a future version of Ginkgo.  This feature has proved to be more noisy than useful.  You can use --poll-progress-after, instead, to get more actionable feedback about potentially slow specs and understand where they might be getting stuck.

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=2.19.0


Ginkgo ran 1 suite in 52.886365332s
Test Suite Passed
You're using deprecated Ginkgo functionality:
=============================================
  --debug is deprecated
  Learn more at: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#removed--debug

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=2.1.6

root@devVm1:~/workspace/src/github.com/IBM/ibm-object-csi-driver# 
```

File: controllers/syncer/csi_node.go [hunk 1]
```diff
@@ -44,6 +44,13 @@ func NewCSINodeSyncer(c client.Client, driver *crutils.IBMObjectCSI) syncer.Inte
 				},
 				Spec: corev1.PodSpec{},
 			},
+			MinReadySeconds: 30,
+			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+				Type: appsv1.RollingUpdateDaemonSetStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+					MaxUnavailable: func(i intstr.IntOrString) *intstr.IntOrString { return &i }(intstr.FromString("100%")),
```
Comment (Bhagyashreek8) [inline controllers/syncer/csi_node.go:51]: are we setting any value for `MaxUnavailable:` @ambiknai ?

Comment (ambiknai) [reply controllers/syncer/csi_node.go:51]: yes we are setting 100%

**Review by Bhagyashreek8 — APPROVED** [review]
lgtm

File: controllers/syncer/csi_node.go (hunk 1 above)
Comment (ambiknai) [reply controllers/syncer/csi_node.go]: Fixed the typo. Thanks @Bhagyashreek8

**Review by ashimagarg27 — APPROVED** [review]

**Pull request merged**